indexing-hadoop/src/main/java/io/druid/indexer/JobHelper.java

```diff
@@ -331,7 +331,9 @@ public static boolean runJobs(List<Jobby> jobs, HadoopDruidIndexerConfig config)
         Path workingPath = config.makeIntermediatePath();
         log.info("Deleting path[%s]", workingPath);
         try {
-          workingPath.getFileSystem(injectSystemProperties(new Configuration())).delete(workingPath, true);
+          if( workingPath.getFileSystem(injectSystemProperties(new Configuration())).delete(workingPath, true) ){
+            existing.clear();
```

Comment:
Are you sure the `runJobs` only run once in a single jvm?
For a hadoop-index task have two phases, determine-partion and index-generation
From the tasks log, two jobs use the same cache directory on hdfs.
https://github.com/druid-io/druid/files/187650/file_not_found_chg.txt

Different hadoop-index task uses different directories, I can see that from log
```
log1: 
 Uploading jar to path[/tmp/druid-indexing/myds/2016-03-24T171908.371+0800/5704cb69549d4d499288899e39273825/classpath/druid-aws-common-0.9.1-SNAPSHOT.jar]

log2: 
Uploading jar to path[/tmp/druid-indexing/myds/2016-03-24T175310.279+0800/e5e280a57eb548a79db3a4770723e364/classpath/druid-aws-common-0.9.1-SNAPSHOT.jar]
```